lib/breadcrumbs.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+var BreadcrumbBuffer = function BreadcrumbBuffer(maxBreadcrumbs) {
+  this._buffer = [];
+  this._maxBreadcrums = maxBreadcrumbs;
+};
+
+BreadcrumbBuffer.prototype.add = function add(crumb) {
+  this._buffer.push(crumb);
+  if (this._buffer.length > this._globalOptions.maxBreadcrumbs) {
+    this._buffer.shift();
```

Comment:
Array#shift is O(n), n is small here but if we end up with a nice way to avoid shift let's avoid it; this is basically a ringbuffer so maybe we can ref some JS data structure implementation libraries to see if there's a better way. Probably not worth just busting out a deque implementation though.